### DISCUSSION TITLE
Change of icon to make it more explicit of what its action will be

### DISCUSSION THREAD

**Discussion by portusgallus** [home-assistant/frontend#18556] [Accessibility]
I don't know how to properly classify this one as it's simply an image swap for an icon.  In ToDo lists, checked items may be deleted/purged from the list by clicking an action button that uses an icon with the appearance of a hierarchy list.  This icon, to my family's eyes, tells them the list will be sorted or it will give them an opportunity to re-order them manually, which they desperately want to do in the checked items, however its function is instead to delete all the checked items on the list, which makes us have to start over again for maintaining a static frequent-purchase items list.  

I would like to request that the purge button have its icon changed from a list-hierarchy image to a more classical trashcan image that more closely matches our historically trained expected function of that button.  This will save me having to restore the .json file from a backup every time (prior to this month's update anyway, the .json fix no longer gracefully takes care of this for me).

Thank you.

**Comment by karwosts** [comment]
I'll refrain from proposing an alternate icon, since I don't know the history of the current one (it is named `notificationClearAll`, so there seems like there is some precedent for this being a clearing action), but I do believe the general frontend policy here is that anything destructive should have a confirmation dialog with explanation, which isn't present here. I'll see if I can get one added. 

I think that should address your concern?

**Reply by piitaya** [reply]
Just a proposal but what about changing the icon to something like that ? https://pictogrammers.com/library/mdi/icon/delete-sweep/
![CleanShot 2023-12-05 at 09 08 37](https://github.com/home-assistant/frontend/assets/5878303/f5ba3ae6-3b4d-48fc-9b73-3d571c6ec205)

**Reply by karwosts** [reply]
Icon looks good to me.

#18905 adds the confirmation dialog.

**Reply by portusgallus** [reply]
I like the delete-sweep icon, it should do a much better job of preserving my sanity.  My mother is 80 and usually insists madly that if she has to read one more popup dialog the globe will suddenly disintegrate into a ball of fire.  :)  Yes, i reserve at least three hours per week to clean up her pc.  I do like the confirmation dialog option though, it would be great for the usual user.
